python/src/equistore/operations/slice.py

```diff
@@ -0,0 +1,395 @@
+import warnings
+
+import numpy as np
+
+from ..tensor import TensorMap
+from ..block import TensorBlock
+from ..labels import Labels
+
+
+def slice(tensormap: TensorMap, samples_to_slice=None, properties_to_slice=None):
+    """
+    Slices an input :py:class:`TensorMap` along the samples and/or properties
+    dimension(s). `samples_to_slice` and `properties_to_slice` are
+    :py:class:`Labels` objects that specify the samples/properties
+    (respectively) names and indices that should be sliced, i.e. kept in the
+    output tensor.
+
+    Note that either `samples_to_slice` or `properties_to_slice`, or both,
+    should be specified as input. User warnings will be issued if any returned
+    blocks are empty after slicing.
+
+    .. code-block:: python
+        sliced_tensormap = slice(
+            tensormap,
+            samples=Labels(
+                names=["structure", "center"],
+                values=np.array(
+                    [[0, 1], [0, 6], [1, 6], [3, 16]]
+                ),  # must be a 2D-array
+            )
+            properties=Labels(
+                names=["n",],  # radial channel
+                values=np.array([[3,], [4,], [5,]]),
+            )
+        )
+
+    Also note that this function will return a :py:class:`TensorMap` whose
+    blocks are of equal or smaller dimensions (due to slicing) than those of the
+    input. However, the returned :py:class:`TensorMap` will be returned with the
+    same number of blocks and the corresponding keys as in the input. If any
+    block upon slicing is reduced to nothing, i.e. in the case that it has none
+    of the specified `samples_to_slice` or `properties_to_slice`, an empty block
+    will be returned but will still be accessible by its key. User warnings will
+    be issued if any blocks are sliced to contain no values.
+
+    For the empty blocks that may be returned, although there will be no actual
+    values in its TensorBlock.values tensor, the shape of this tensor will be
+    non-zero in the dimensions that haven't been sliced. This is created by
+    slicing the input TensorBlock, as opposed to just returning an
+    artificially-created empty one (with no shape or dimensions), and is
+    intentional. It allows the slicing of dimensions to be tracked.
```

Comment:
```suggestion
    non-zero in the dimensions that haven't been sliced. This allow slicing of dimensions to be tracked.
```

This part of the docs feels a bit verbose, how about cutting them down a bit?